crates/sui-types/src/lib.rs

```diff
@@ -196,10 +198,15 @@ pub fn is_primitive(
 
         S::StructInstantiation(idx, targs) => {
             let resolved_struct = resolve_struct(view, *idx);
-            // is option of a primitive
-            resolved_struct == RESOLVED_STD_OPTION
+            // option is a primitive
+            (resolved_struct == RESOLVED_STD_OPTION
                 && targs.len() == 1
-                && is_primitive(view, function_type_args, &targs[0])
+                && is_primitive(view, function_type_args, &targs[0])) ||
+                // `Receiving` struct in the `sui::transfer` module is also a primitive
+                (
+                    resolved_struct == RESOLVED_RECEIVING_STRUCT
+                    && targs.len() == 1
+                )
```

Comment:
This function `is_primitive` is generally to decide what types have values that can be created via `CallArg::Pure`, i.e. directly from BCS bytes. 
So I wouldn't have expected this to be here. What issue did you run into with this and can we remove it?

Reply:
It's gone! I think the issue I originally had was I didn't add it properly into the sui mode/verification initially, but now that I have this can be removed.